DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -218,58 +186,47 @@ CentreAnswersData centreAnswersData
                 myAccountDetailsData.ProfileImage,
                 myAccountDetailsData.ProfessionalRegistrationNumber,
                 myAccountDetailsData.HasBeenPromptedForPrn,
-                myAccountDetailsData.JobGroupId!.Value,
+                myAccountDetailsData.JobGroupId,
                 detailsLastChecked,
-                userId
+                myAccountDetailsData.UserId
             );
 
-            userDataService.SetCentreEmail(
-                userId,
-                centreAnswersData.CentreId,
-                centreAnswersData.CentreEmail
-            );
+            if (centreId != null)
+            {
+                userDataService.SetCentreEmail(
+                    myAccountDetailsData.UserId,
+                    centreId.Value,
+                    centreEmail
+                );
+            }
 
-            if (delegateUsers.Count != 0)
+            if (delegateDetailsData != null)
             {
-                var delegateId = myAccountDetailsData.DelegateId!.Value;
                 // TODO HEEDLS-887 Remove this call once groupsService.SynchroniseUserChangesWithGroups is refactored

```

Comment:
If we're ever going to refactor that method, now is probably the time to do it.

Reply:
Just working out if we actually have sufficient motivation to.

Reply:
I don't think we do actually. Even if we do refactor it to use a `Delegate` rather than a `DelegateUser` you'll still have to call the db to get the delegate's old registration field responses. And this line is all it would change (along with the line in the other place it's called I guess).